x/participation/keeper/msg_participate_test.go

```diff
@@ -34,23 +41,72 @@ func Test_msgServer_Participate(t *testing.T) {
 	require.NotNil(t, res.BaseAuction)
 	auctionID := res.BaseAuction.Id
 
+	addrWithDels := sample.Address()
+	tk.DelegateN(sdkCtx, addrWithDels, 100, 10)
+	availableAlloc, err := tk.ParticipationKeeper.GetAvailableAllocations(sdkCtx, addrWithDels)
+	require.NoError(t, err)
+
 	tests := []struct {
-		name string
-		msg  *types.MsgParticipate
-		err  error
+		name                  string
+		msg                   *types.MsgParticipate
+		desiredUsedAlloc      uint64
+		currentAvailableAlloc uint64
+		err                   error
+		untypedErr            bool
 	}{
 		{
-			name: "should allow to add the participant as allowed bidder in the auction",
+			name: "valid message",
+			msg: &types.MsgParticipate{
+				Participant: addrWithDels,
+				AuctionID:   auctionID,
+				TierID:      1,
+			},
+			desiredUsedAlloc:      1,
+			currentAvailableAlloc: availableAlloc,
+		},
+		{
+			name: "should prevent if double bid from same address",
```

Comment:
```suggestion
			name: "should prevent participating twice in the same auction",
```
There is no bid before auction start